packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1506 @@
+      platform: "node",
+      external: ["aws-sdk"],
+      outfile: path.join(outputPath, "server.js"),
+      // Need to add the --ignore-annotations flag to ESBuild. It appears Remix templates by default include the "sideEffects": false, flag in their package.json. This causes ESBuild to remove the required polyfill module. I had removed the flag so it was working fine for me, but likely we will want to avoid having everyone update their Remix app like this. Remix includes polyfills so it is always going to be by nature with side effects. I've asked on their GH the reasoning for the flag being there in the first place. There is also an active PR to remove the need for this flag in their package.json, although the PR doesn't provide the original motivation for the flag being there. I will do more research and get back to ya'll. For sure we will want to address this. https://esbuild.github.io/api/#ignore-annotations
+      ignoreAnnotations: true,
+    })
+
+    if (result.errors.length > 0) {
+      result.errors.forEach((error) => console.error(error));
+      throw new Error(`There was a problem bundling the server.`);
+    }
+
+    return outputPath;
+  }
+
+  private createServerLambdaBundleWithStub(): string {
+    // Use existing stub bundle in assets
+    return path.resolve(__dirname, "../assets/RemixSite/server-lambda-stub");
+  }
+
+  private createServerFunctionRoleForRegional(): iam.Role {
+    return this.createServerFunctionRole([
+      "lambda.amazonaws.com",
+    ]);
+  }
+
+  private createServerFunctionRoleForEdge(): iam.Role {
+    return this.createServerFunctionRole([
+      "lambda.amazonaws.com",
+      "edgelambda.amazonaws.com",
+    ]);
+  }
+
+  private createServerFunctionRole(principals: string[]): iam.Role {
+    const { defaults } = this.props;
+
+    // Create function role
+    const role = new iam.Role(this, `ServerLambdaRole`, {
+      assumedBy: new iam.CompositePrincipal(
+        ...principals.map((principal) => new iam.ServicePrincipal(principal))
+      ),
+      managedPolicies: [
+        iam.ManagedPolicy.fromManagedPolicyArn(
+          this,
+          "EdgeLambdaPolicy",
+          "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+        ),
+      ],
+    });
+
+    // Attach permission
+    this.cdk.bucket.grantReadWrite(role);
+    if (defaults?.function?.permissions) {
+      attachPermissionsToRole(role, defaults.function.permissions);
+    }
+
+    return role;
+  }
+
+  private createServerFunctionForRegional(bundlePath: string): lambda.Function {
+    const { defaults, environment } = this.props;
+
+    return new lambda.Function(this, `ServerFunction`, {
+      description: "Server handler for Remix",
+      handler: "server.handler",
+      currentVersionOptions: {
+        removalPolicy: RemovalPolicy.DESTROY,
+      },
+      logRetention: logs.RetentionDays.THREE_DAYS,
+      code: lambda.Code.fromAsset(bundlePath),
+      runtime: lambda.Runtime.NODEJS_16_X,
```

Comment:
**question:** We have Node16 statically configured everywhere, I wonder if it is worth calling that out in the docs explicitly?